### PR TITLE
Added integration with CircleCI and service_pull_request 

### PR DESCRIPTION
**Pull request by t-3-k**
Adde some CircleCI specific integration plus support for pull request comments.
API post parameter service_pull_request is the associated pull request ID of the build. It is used for updating the status and/or commenting.

### CODE REVIEW

File: coveralls/api.py [hunk 1]
```diff
@@ -47,18 +47,25 @@ def __init__(self, token_required=True, **kwargs):
             self.config['repo_token'] = repo_token
 
         if os.environ.get('TRAVIS'):
-            is_travis = True
+            is_travis_or_circle = True
             self.config['service_name'] = file_config.get('service_name', None) or 'travis-ci'
             self.config['service_job_id'] = os.environ.get('TRAVIS_JOB_ID')
+        elif os.environ.get('CI'):
```
Comment (danpalmer) [inline coveralls/api.py]: Should this be `CIRCLE` instead of `CI`? Might be a bit confusing otherwise, as Jenkins is a perfectly valid *CI* to be using.

**Comment by danpalmer** [comment]
Can I do anything to help get this merged, would be a really handy feature to have.

**Comment by coagulant** [comment]
Having regressin tests would be nice.

Atm, Travis build will fail because of #81 and #63
You can manually check coverage3 tests versions passing using `tox`

File: coveralls/api.py (hunk 1 above)
Comment (t-3-k) [inline coveralls/api.py]: You are right about that, changed CI to CIRCLECI

**Pull request closed without merging**